// grifter-js/grifter/models/rtb/index.js


var enums = require('./enums');
var models = require('./models');


// Export Enums
module.exports.AdPosition             = enums.AdPosition;
module.exports.APIFrameworks          = enums.APIFrameworks;
module.exports.BannerAdType           = enums.BannerAdType;
module.exports.CompanionType          = enums.CompanionType;
module.exports.ConnectionType         = enums.ConnectionType;
module.exports.CreativeAttribute      = enums.CreativeAttribute;
module.exports.ContentCategory        = enums.ContentCategory;
module.exports.ContentDeliverMethod   = enums.ContentDeliverMethod;
module.exports.ContentContext         = enums.ContentContext;
module.exports.DeviceType             = enums.DeviceType;
module.exports.ExpandableDirection    = enums.ExpandableDirection;
module.exports.IPLocationServices     = enums.IPLocationServices;
module.exports.IQGMediaRating         = enums.IQGMediaRating;
module.exports.NoBidReasonCode        = enums.NoBidReasonCode;
module.exports.ProductionQuality      = enums.ProductionQuality;
module.exports.Protocols              = enums.Protocols;




// Import Models
module.exports.App           = models.app.schema;
module.exports.Audio         = models.audio.schema;
module.exports.Banner        = models.banner.schema;
module.exports.Bid           = models.bid.schema;
module.exports.BidResponse   = models.bidresponse.schema;
module.exports.BidRequest    = models.bidrequest.schema;
module.exports.Content       = models.content.schema;
module.exports.Data          = models.data.schema;
module.exports.Imp           = models.imp.schema;
module.exports.Native        = models.native.schema;
module.exports.Pmp           = models.pmp.schema;
module.exports.Producer      = models.producer.schema;
module.exports.Publisher     = models.publisher.schema;
module.exports.Regs          = models.regs.schema;
module.exports.SeatBid       = models.seatbid.schema;
module.exports.Segment       = models.segment.schema;
module.exports.Video         = models.video.schema;

module.exports.enums  = enums;
module.exports.models = models;
