// grifter-js/grifter/models/models_index.js

// Third party modules
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var enums = require('./enums');

// console.log(enums);


exports.configure = function(callback){
  callback();
}

// Import Models
var App           = require('./models/app').schema;
var Audio         = require('./models/audio').schema;
var Banner        = require('./models/banner').schema;
var Bid           = require('./models/bid').schema;
var BidRequest    = require('./models/bid_request').schema;
var BidResponse   = require('./models/bid_response').schema;
var Content       = require('./models/content').schema;
var Data          = require('./models/data').schema;
var Format        = require('./models/format').schema;
var Imp           = require('./models/imp').schema;
var Native        = require('./models/native').schema;
var Pmp           = require('./models/pmp').schema;
var Producer      = require('./models/producer').schema;
var Publisher     = require('./models/publisher').schema;
var Regs          = require('./models/regs').schema;
var SeatBid       = require('./models/seatbid').schema;
var Segment       = require('./models/segment').schema;
var Video         = require('./models/video').schema;


// Export Enums
module.exports.AdPosition             = enums.AdPosition;
module.exports.APIFrameworks          = enums.APIFrameworks;
module.exports.BannerAdType           = enums.BannerAdType;
module.exports.CompanionType          = enums.CompanionType;
module.exports.ConnectionType         = enums.ConnectionType;
module.exports.CreativeAttribute      = enums.CreativeAttribute;
module.exports.ContentCategory        = enums.ContentCategory;
module.exports.ContentDeliverMethod   = enums.ContentDeliverMethod;
module.exports.ContentContext         = enums.ContentContext;
module.exports.DeviceType             = enums.DeviceType;
module.exports.ExpandableDirection    = enums.ExpandableDirection;
module.exports.IPLocationServices     = enums.IPLocationServices;
module.exports.IQGMediaRating         = enums.IQGMediaRating;
module.exports.NoBidReasonCode        = enums.NoBidReasonCode;
module.exports.ProductionQuality      = enums.ProductionQuality;
module.exports.Protocols              = enums.Protocols;




// Export Models
module.exports.App           = App;
module.exports.Audio         = Audio;
module.exports.Banner        = Banner;
module.exports.Bid           = Bid;
module.exports.BidResponse   = BidResponse;
module.exports.BidRequest    = BidRequest;
module.exports.Content       = Content;
module.exports.Data          = Data;
module.exports.Imp           = Imp;
module.exports.Native        = Native;
module.exports.Pmp           = Pmp;
module.exports.Producer      = Producer;
module.exports.Publisher     = Publisher;
module.exports.Regs          = Regs;
module.exports.SeatBid       = SeatBid;
module.exports.Segment       = Imp;
module.exports.Video         = Video;
