// grifter-models/lib/grifter_campaign.js
/**
Account
*/

module.exports = function  (sequelize, DataTypes) {
  var Campaign = sequelize.define('Campaign',
  {
    campaignId:       { primaryKey: true, type: DataTypes.UUID, defaultValue: DataTypes.UUIDV4 },
    title:            { type:DataTypes.STRING, allowNull:false },
    description:      { type:DataTypes.STRING, allowNull:false },
    brand:            { type:DataTypes.STRING, allowNull:false },
    startDate:        { type:DataTypes.DATE, allowNull:false },
    endDate:          { type:DataTypes.DATE, allowNull:false },
    budgetCurrency:   { type:DataTypes.STRING, allowNull: false, default: "USD" },
    budgetFund:       { type:DataTypes.DOUBLE, allowNull:false, default: 0 },
    active:			  { type:DataTypes.BOOLEAN, allowNull:false, default: true }
  },
  {
    classMethods:{
      associate: function (models) {
        Campaign.belongsTo(models.Account);
        Campaign.hasMany(models.Ad);
        Campaign.hasMany(models.Target);
        Campaign.hasMany(models.Reciept);
      }
    }
  });
  return Campaign;
}
