// grifter-models/lib/grifter_ad_banner.js
/**
Banner
*/

module.exports = function  (sequelize, DataTypes) {
  var Banner = sequelize.define('Banner',
  {
    id:           { type: DataTypes.UUID,primaryKey: true, defaultValue: DataTypes.UUIDV4 },
    title:        { type: DataTypes.STRING, defaultValue:"Untitled Banner"},
    description:  { type: DataTypes.STRING, defaultValue:"No Description" },
    mimes:        { type: DataTypes.ARRAY( DataTypes.STRING ), defaultValue:[] },
    w:            { type: DataTypes.INTEGER, defaultValue:0 },
    h:            { type: DataTypes.INTEGER, defaultValue:0 },
    pos:          { type: DataTypes.INTEGER },
    expdir:       { type: DataTypes.INTEGER },
    api:          { type: DataTypes.INTEGER },
    btype:        { type: DataTypes.ARRAY( DataTypes.INTEGER )},
    topframe:     { type: DataTypes.BOOLEAN },
    api:          { type: DataTypes.ARRAY( DataTypes.INTEGER )},
    battr:        { type: DataTypes.ARRAY( DataTypes.INTEGER )},
  },
  {
    classMethods:{
      associate: function (models) {
        Banner.belongsTo(models.Ad, { allowNull:false });
      }
    }
  });
  return Banner;
}
