// grifter-models/lib/grifter_ad_audio.js
/**
Ad
*/

module.exports = function  (sequelize, DataTypes) {
  var Audio = sequelize.define('Audio',
  {
    id:           { type: DataTypes.UUID,primaryKey: true, defaultValue: DataTypes.UUIDV4 },
    title:        { type: DataTypes.STRING },
    description:  { type: DataTypes.STRING },
    mimes:        { type: DataTypes.ARRAY( DataTypes.STRING ) },
    minduration:  { type: DataTypes.INTEGER },
    maxduration:  { type: DataTypes.INTEGER },
    startdelay:   { type: DataTypes.INTEGER },
    sequence:     { type: DataTypes.INTEGER },
    maxextended:  { type: DataTypes.INTEGER },
    minbitrate:   { type: DataTypes.INTEGER },
    maxbitrate:   { type: DataTypes.INTEGER },
    maxseq:       { type: DataTypes.INTEGER },
    feed:         { type: DataTypes.INTEGER },
    stitched:     { type: DataTypes.INTEGER },
    nvol:         { type: DataTypes.INTEGER },
    stitched:     { type: DataTypes.BOOLEAN },
    api:          { type: DataTypes.INTEGER },
    protocols:    { type: DataTypes.ARRAY( DataTypes.INTEGER )},
    battr:        { type: DataTypes.ARRAY( DataTypes.INTEGER )},
    delivery:     { type: DataTypes.ARRAY( DataTypes.INTEGER )},
    companiontype:{ type: DataTypes.ARRAY( DataTypes.INTEGER )}
  },
  {
    classMethods:{
      associate: function (models) {
        Audio.belongsToMany(models.Ad, { through:"audio_ads", allowNull:false })
        Audio.hasOne(models.Banner)
      }
    }
  });
  return Audio;
}
