// grifter-models/lib/grifter_ad_video.js
/**
Video
*/


module.exports = function  (sequelize, DataTypes) {
  var Video = sequelize.define('Video',
  {
    id:             { type: DataTypes.UUID,primaryKey: true, defaultValue: DataTypes.UUIDV4 },
    title:          { type:DataTypes.STRING },
    description:    { type:DataTypes.STRING },
    mimes:          { type: DataTypes.ARRAY( DataTypes.STRING ) },
    minduration:    { type: DataTypes.INTEGER },
    maxduration:    { type: DataTypes.INTEGER },
    protocols:      { type: DataTypes.ARRAY( DataTypes.INTEGER )},
    w:              { type: DataTypes.INTEGER },
    h:              { type: DataTypes.INTEGER },
    startdelay:     { type: DataTypes.INTEGER },
    linearity:      { type: DataTypes.INTEGER },
    skip:           { type: DataTypes.BOOLEAN },
    skipmin:        { type: DataTypes.INTEGER },
    skipafter:      { type: DataTypes.INTEGER },
    sequence:       { type: DataTypes.INTEGER },
    battr:          { type: DataTypes.ARRAY( DataTypes.INTEGER )},
    maxextended:    { type: DataTypes.INTEGER },
    minbitrate:     { type: DataTypes.INTEGER },
    maxbitrate:     { type: DataTypes.INTEGER },
    boxingallowed:  { type: DataTypes.BOOLEAN },
    playbackmethod: { type: DataTypes.INTEGER }
  },
  {
    classMethods:{
      associate: function (models) {
        Video.belongsToMany(models.Ad, { through:"video_ads", allowNull:false })
      }
    }
  });
  return Video;
}
