// grifter-js/grifter/adapters/mopub/handle_bid.js
var _ = require('underscore');
var models = require('../../models').rtb;
var mopub = require('./index');
var logging = require('../../logging');

module.exports = function(req, res){
  // console.log(req.method);
  console.log("==============================\nBID REQUEST: Mopub");
  console.log(req.body);
  console.log("==============================\n");
  // console.log(req.query);

  if(req.method == "POST") {
    handlePOST(req,res);
  }else if(req.method == "GET"){
    handleGET(req,res);
  }
}

function handleGET (req, res) {
  console.log("handle GET");
  handleBidRequest( req, res );
}

function handlePOST (req, res) {
  console.log("handle POST");
  handleBidRequest( req, res );
}



function handleBidRequest (req, res) {
  var bidrequest = new models.BidRequest(req.body);
  console.log("> BID ID:", bidrequest.id);
  console.log("> AT:", bidrequest.at);

  // Device
  if(bidrequest.device){
    var device = bidrequest.device;
    console.log("Device:");
    console.log(" - make:", device.make);
    console.log(" - model:", device.model);
    console.log(" - os:", device.os);
    console.log(" - osv:", device.osv);
    console.log(" - ua:", device.ua);
    console.log(" - devicetype:", models.DeviceType.get(device.devicetype).key);
    console.log(" - geo:", device.geo.country, device.geo.lat, device.geo.lon);
  }else{
    console.log("No Device");
  }

  // User
  if(bidrequest.user){
    var user = bidrequest.user;
    console.log("User:");
    console.log(" - gender:", user.gender);
    console.log(" - yob:", user.yob);
    console.log(" - customdata:", JSON.stringify(user.customdata));
    console.log(" - geo:", JSON.stringify(user.geo));
    console.log(" - data:", JSON.stringify(user.data));
    console.log(" - buyeruid:", user.buyeruid) ;
  }else{
    console.log("No User");
  }


  // Site or App
  if( bidrequest.site ){
    // Site
    var site = bidrequest.site;
    console.log("Site:", site.name);
    console.log(" - id:", site.id);
    console.log(" - categrories:", JSON.stringify(site.cat));
    console.log(" - domain:", site.domain);
    console.log(" - publisher:", JSON.stringify(site.publisher));
  }else if( bidrequest.app ){
    // App
    var app = bidrequest.app;
    console.log("App:", app.name);
    console.log(" - id:", app.id);
    console.log(" - categrories:", app.cat);
    console.log(" - domain:", app.domain);
  }else{
    // unknown
    console.log("Unknown Requester");
  }

   // Ext
  if(bidrequest.ext){
    var ext = bidrequest.ext;
    console.log("EXT:", ext);
  }else{
    console.log("No EXT");
  }

  // Imp
  for (var i in bidrequest.imp){
    var imp = bidrequest.imp[i];
    console.log(" >> id:", imp.id);
    console.log(" >> displaymanager:", imp.displaymanager);
    console.log(" >> instl:", imp.instl);
    console.log(" >> bidfloor: ", imp.bidfloor );
    if(imp.banner.w > 0) {
      console.log("banner impression");
      var b = imp.banner;
      console.log(" >>> w:", b.w);
      console.log(" >>> h:", b.h);
      console.log(" >>> mimes:", JSON.stringify(b.mimes));
      var btypes = b.btype.map(function (x) {
        return models.BannerAdType.get(x).key;
      });
      console.log(" >>> btype:", btypes);

      /// respond for banner request here:
      var price = 0.52;
      var adm = admarkup(b.w, b.h);
      var ext =  mopub.createBidExt( b.btype, price );

      var bid_data = {
        id:"foobar",
        impid:"testimpid",
        price:price,
        adid:"testadid",
        nurl:"http://rtb-grifter.rhcloud.com/win",
        adomain:["apple.com"],
        bundle:"com.apple.itunes.freemusic1",
        iurl: "https://s3-us-west-2.amazonaws.com/polyimages/mobfox-320x50_1.jpg",
        cid:"testcampaignid",
        crid:"testcreativeid",
        cat:["IAB1-6"],
        attr:[0],
        dealid:"testdealid",
        h: b.h,
        w: b.w,
        ext:ext,
        adm: adm
      };

      var seatbid_data = {
        bid: [ new models.Bid( bid_data ) ],
        seat:"mytestseat",
        group: 0
      };

      var response_data = {
        id:"abc123",
        bidid:"mytestbidid",
        cur:"USD",
        nbr:0,
        seatbid:[new models.SeatBid(seatbid_data)]
      };

      logging.bid_requested( bidrequest.id, bid_data.price, "mopub");

      var bid_response = new models.BidResponse( response_data );
      res.setHeader('Content-Type', 'application/json');
      res.send(JSON.stringify(bid_response));
      ///
    } else if(imp.video) {
      console.log("video impression");
      console.log(imp.video);
    } else if(audio) {
      console.log("audio impression");
    } else if(imp.native) {
      console.log("native impression");
    }
  }


}


var admarkup = function (w, h) {
  console.log("Generating Ad Markup for bid from MOPUB");
  return "";
}


var richmedia_mimes = [ "application/javascript", "text/javascript"];
var text_mimes = [ "text/html","text/plain" ];
var image_mimes = [ "image/gif", "image/jpeg", "image/png" ];
