// grifter-js/grifter/models/rtb/enums/index.js

var Enum = require('enum');

module.exports.AdPosition                 = require('./ad_position').AdPosition;
module.exports.APIFrameworks              = require('./api_frameworks').APIFrameworks;
module.exports.BannerAdType               = require('./banner_ad_type').BannerAdType;
module.exports.CompanionType              = require('./companion_type').CompanionType;
module.exports.ConnectionType             = require('./connection_type').ConnectionType;
module.exports.CreativeAttribute          = require('./creative_attribute').CreativeAttribute;
module.exports.ContentCategory            = require('./content_category').ContentCategory;
module.exports.ContentDeliverMethod       = require('./content_deliver_method').ContentDeliverMethod;
module.exports.ContentContext             = require('./content_context').ContentContext;
module.exports.DeviceType                 = require('./device_type').DeviceType;
module.exports.ExpandableDirection        = require('./expandable_direction').ExpandableDirection;
module.exports.FeedType                   = require('./feed_type').FeedType;
module.exports.IPLocationServices         = require('./ip_location_services').IPLocationServices;
module.exports.IQGMediaRating             = require('./iq_media_rating').IQGMediaRating;
module.exports.LocationType               = require('./location_type').LocationType;
module.exports.NoBidReasonCode            = require('./no_bid_response_code').NoBidReasonCode;
module.exports.ProductionQuality          = require('./production_quality').ProductionQuality;
module.exports.Protocols                  = require('./protocols').Protocols;
module.exports.VideoLinearity             = require('./video_linearity').VideoLinearity;
module.exports.VolumeNormalizationMode    = require('./volume_normalization_mode').VolumeNormalizationMode;


Enum.prototype.keys = function() {
  return this.enums.map(function (e) { return e.key; });
};

Enum.prototype.values = function() {
  return this.enums.map(function (e) { return e.value; });
};
