// grifter-js/grifter/models/rtb/models/app.js
var SchemaObject = require('schema-object');

/**
App
This object should be included if the ad supported content is a non-browser application (typically in mobile) as opposed to a website. A bid request must not contain both an App and a Site object. At a minimum, it is useful to provide an App ID or bundle, but this is not strictly required.
*/


var Publisher = require('./publisher');
var Content = require('./content');
var ContentCategory = require('../enums').ContentCategory;

var category_enums = ContentCategory.keys();

var definition = {
  id:             { type: String },
  name:           { type: String },
  domain:         { type: String },
  bundle:         { type: String },
  storeurl:       { type: String },
  ver:            { type: String },
  cat:            [ { type : String, enum: category_enums } ],
  sectioncat:     [ { type : String, enum: category_enums } ],
  pagecat:        [ { type : String, enum: category_enums } ],
  keywords:       { type: String },
  privacypolicy:  { type: Boolean },
  paid:           { type: Boolean },
  publisher:      { type: Publisher.definition },
  content:        { type: Content.definition },
  ext:            'any'
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);;
