// grifter-js/grifter/models/rtb/models/bid.js
var SchemaObject = require('schema-object');


/**
 # Bid
 A bid response can contain multiple SeatBid objects, each on behalf of a different bidder seat and each containing one or more individual bids. If multiple impressions are presented in the request, the group attribute can be used to specify if a seat is willing to accept any impressions that it can win (default) or if it is only interested in winning any if it can win them all as a group.*/
var enums = require('../enums');



var definition = {
  id:               { type: String, required: true },
  impid:            { type: String, required: true },
  adid:             { type: String, required: true },
  nurl:             { type: String, required: true },
  adm:              { type: String },
  bundle:           { type: String },
  adomain:          { type: String },
  iurl:             { type: String },
  cid:              { type: String },
  crid:             { type: String },
  dealid:           { type: String },
  cat:              [ { type : String, enum: enums.ContentCategory.keys() } ],
  var:              [ { type: String } ],
  price:            { type: Number },
  api:              { type : String, enum: enums.APIFrameworks.keys() },
  protocol:         { type : String, enum: enums.APIFrameworks.keys() },
  qagmediarating:   { type: Number },
  w:                { type: Number },
  h:                { type: Number },
  exp:              { type: Number },
  attr:             [ { type : Number, enum: enums.CreativeAttribute.values() } ],
  ext:              { type: 'any' }
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);;
