// grifter-js/grifter/models/rtb/models/bid_response.js
var SchemaObject = require('schema-object');

var Publisher = require('./publisher');
var Content   = require('./content');
var SeatBid   = require('./seatbid');
var enums = require('../enums');
/**
 BidResponse
 This object is the top-level bid response object (i.e., the unnamed outer JSON object). The id attribute is a reflection of the bid request ID for logging purposes. Similarly, bidid is an optional response tracking ID for bidders. If specified, it can be included in the subsequent win notice call if the bidder wins. At least one seatbid object is required, which contains at least one bid for an impression. Other attributes are optional.
 To express a “no-bid”, the options are to return an empty response with HTTP 204. Alternately if the bidder wishes to convey to the exchange a reason for not bidding, just a BidResponse object is returned with a reason code in the nbr attribute.
*/

var definition = {
  id:             { type: String, required: true },
  seatbid:        [ { type:SeatBid.definition, minLength: 1 } ],
  bidid:          { type: String },
  cur:            { type : String, default : "USD" },
  customdata:     { type: String },
  nbr:            { type: Number, enum: enums.NoBidReasonCode.values() },
  ext:            { type: 'any' }
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);;
