// grifter-js/grifter/models/rtb/models/bid_request.js
var SchemaObject = require('schema-object');

var App     = require('./app');
var Device  = require('./device');
var Imp     = require('./imp');
var Regs    = require('./regs');
var Site    = require('./site');
var User    = require('./user');



/**
BidRequest
The top-level bid request object contains a globally unique bid request or auction ID. This id attribute is required as is at least one impression object (Section 3.2.2). Other attributes in this top-level object establish rules and restrictions that apply to all impressions being offered.
There are also several subordinate objects that provide detailed data to potential buyers. Among these are the Site and App objects, which describe the type of published media in which the impression(s) appear. These objects are highly recommended, but only one applies to a given bid request depending on whether the media is browser-based web content or a non-browser application, respectively.
*/

var definition = {
  id:       { type: String, required: true, unique: true },
  wseat:    [ { type: String } ],
  cur:      [ { type: String } ],
  bcat:     [ { type: String } ],
  badv:     [ { type: String } ],
  bapp:     [ { type: String } ],
  imp:      [ { type: Imp.definition, minLength: 1 } ],
  site:     { type: Site.definition },
  app:      { type: App.definition },
  device:   { type: Device.definition },
  user:     { type: User.definition },
  test:     { type: Boolean },
  at:       { type: Number },
  tmax:     { type: Number },
  allimps:  { type: Number },
  regs:     { type: Regs.definition },
  ext:      { type: 'any' }
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);;
