// grifter-js/grifter/models/rtb/models/content.js
var SchemaObject = require('schema-object');

/**
Content
*/

var Producer = require('./producer');
var Data   = require('./data');
var ContentCategory = require('../enums').ContentCategory;

var category_enums = ContentCategory.keys();

var definition = {
  id:                 { type: String },
  episode:            { type: Number },
  title:              { type: String },
  series:             { type: String },
  season:             { type: String },
  artist:             { type: String },
  genre:              { type: String },
  album:              { type: String },
  isrc:               { type: String },
  url:                { type: String },
  cat:                [ { type : String, enum: category_enums } ],
  producer:           { type: Producer.definition },
  title:              { type: String },
  prodq:              { type: Number },
  context:            { type: String },
  contentrating:      { type: String },
  userrating:         { type: Number },
  qagmediarating:     { type: String },
  keywords:           { type: String },
  livestream:         { type: Boolean },
  sourcerelationship: { type: Boolean },
  len:                { type: Number },
  language:           { type: String },
  embeddable:         { type: Boolean },
  data:               [ { type: Data.definition } ],
  ext:                { type: 'any' }
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);
