// grifter-js/grifter/models/rtb/models/device.js
var SchemaObject = require('schema-object');

/**
 Device
 This object provides information pertaining to the device through which the user is interacting. Device information includes its hardware, platform, location, and carrier data. The device can refer to a mobile handset, a desktop computer, set top box, or other digital device.
 */

var Geo = require('./geo');
var enums = require('../enums');


var definition = {
  ua:               { type: String },
  geo:              { type: Geo.definition },
  dnt:              { type: Boolean },
  lmt:              { type: Boolean },
  ip:               { type: String },
  ipv6:             { type: String },
  devicetype:       { type: Number, enum: enums.DeviceType.values() },
  make:             { type: String },
  model:            { type: String },
  os:               { type: String },
  osv:              { type: String },
  hwv:              { type: String },
  h:                { type: Number },
  w:                { type: Number },
  ppi:              { type: Number },
  pxratio:          { type: Number },
  js:               { type: Boolean },
  geofetch:         { type: Boolean },
  flashver:         { type: String },
  language:         { type: String },
  carrier:          { type: String },
  connectiontype:   { type: Number, enum: enums.ConnectionType.values() },
  ifa:              { type: String },
  didsha1:          { type: String },
  didmd5:           { type: String },
  dpidmd5:          { type: String },
  macsha1:          { type: String },
  macmd5:           { type: String },
  ext:              { type: 'any' }
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);
