// grifter-js/grifter/models/rtb/models/geo.js
var SchemaObject = require('schema-object');

/**
 Geo
 This object encapsulates various methods for specifying a geographic location. When subordinate to a Device object, it indicates the location of the device which can also be interpreted as the user’s current location. When subordinate to a User object, it indicates the location of the user’s home base (i.e., not necessarily their current location).
 Page 21
 OpenRTB API Specification Version 2.4 IAB Technology Lab
 The lat/lon attributes should only be passed if they conform to the accuracy depicted in the type attribute. For example, the centroid of a geographic region such as postal code should not be passed.
*/

var LocationType = require('../enums').LocationType;

var definition = {
  lat:            { type: Number },
  lon:            { type: Number },
  type:           { type: Number, enum: LocationType.values() },
  accuracy:       { type: Number },
  lastfix:        { type: Number },
  ipservice:      { type: Number },
  country:        { type: String },
  region:         { type: String },
  regionfips104:  { type: String },
  metro:          { type: String },
  city:           { type: String },
  zip:            { type: String },
  utcoffset:      { type: Number },
  ext:            { type: 'any' }
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);;
