// grifter-js/grifter/models/rtb/models/imp.js
var SchemaObject = require('schema-object');

var Audio   = require('./audio');
var Video   = require('./video');
var Native  = require('./native');
var Banner  = require('./banner');
var Pmp     = require('./pmp');


/**
Imp
This object describes an ad placement or impression being auctioned. A single bid request can include multiple Imp objects, a use case for which might be an exchange that supports selling all ad positions on a given page. Each Imp object has a required ID so that bids can reference them individually.
The presence of Banner (Section 3.2.3), Video (Section 3.2.4), and/or Native (Section 3.2.6) objects subordinate to the Imp object indicates the type of impression being offered. The publisher can choose one such type which is the typical case or mix them at their discretion. However, any given bid for the impression must conform to one of the offered types.
*/

var definition = {
  id:                 { type: String },
  banner:             { type: Banner.definition },
  video:              { type: Video.definition },
  audio:              { type: Audio.definition },
  native:             { type: Native.definition },
  pmp:                { type: Pmp.definition },
  displaymanager:     { type: String },
  displaymanagerver:  { type: String },
  instl:              { type: Number, default : 0 },
  tagid:              { type: String },
  bidfloor:           { type: Number, default : 0 },
  bidfloorcur:        { type: String, default : "USD" },
  clickbrowser:       { type: Number },
  secure:             { type: Boolean },
  iframebuster:       [ { type: String } ],
  exp:                { type: Number },
  ext:                { type: 'any'}
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);;
