// grifter-js/grifter/models/rtb/models/site.js
var SchemaObject = require('schema-object');

/**
 Site
 This object should be included if the ad supported content is a website as opposed to a non-browser application. A bid request must not contain both a Site and an App object. At a minimum, it is useful to provide a site ID or page URL, but this is not strictly required.
 */

var Publisher = require('./publisher');
var Content = require('./content');
var ContentCategory = require('../enums').ContentCategory;

var category_enums = ContentCategory.keys();

var definition = {
  id:               { type: String },
  name:             { type: String },
  domain:           { type: String },
  cat:              [ { type : String, enum: category_enums } ],
  sectioncat:       [ { type : String, enum: category_enums } ],
  pagecat:          [ { type : String, enum: category_enums } ],
  page:             { type: String },
  ref:              { type: String },
  search:           { type: String },
  mobile:           { type: Boolean },
  privacypolicy:    { type: Boolean },
  publisher:        { type: Publisher.definition },
  content:          { type: Content.definition },
  keywords:         { type: String },
  ext:              { type: 'any' }
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);;
