// grifter-js/grifter/models/rtb/models/video.js
var SchemaObject = require('schema-object');

/**
Video
*/

var VideoLinearity = require('../enums').VideoLinearity;
var Protocols = require('../enums').Protocols;



var definition = {
  mimes:          [ { type: String } ],
  minduration:    { type: Number },
  maxduration:    { type: Number },
  protocols:      [ { type: Number, enum: Protocols.values(), minLength: 1 } ],
  w:              { type: Number },
  h:              { type: Number },
  startdelay:     { type: Number },
  linearity:      { type: Number, enum: VideoLinearity.values() },
  skip:           { type: Boolean },
  skipmin:        { type: Number, default: 0 },
  skipafter:      { type: Number, default: 0 },
  sequence:       { type: Number },
  battr:          [ { type: Number } ],  
  maxextended:    { type: Number },
  minbitrate:     { type: Number },
  maxbitrate:     { type: Number },
  boxingallowed:  { type:Boolean, default:true},
  playbackmethod: [ { type: Number } ],
};

module.exports.definition = definition;
module.exports.schema = new SchemaObject(definition);
