// grifter-js/grifter/logging/index.js
var log_data = {
  entries:[],
  wins:0,
  spend_sum:0.0,
  bid_sum:0.0,
  bid_count:0,
  nobid_count:0

};

var logging = require('grifter-logging');

var Logger = function () {
  this.bid_requested = function (id, price, type) {
    console.log("bid_requested", id, price);
    log_data.bid_sum += price;
    log_data.bid_count += 1;
    log_data.entries.push({"bid":{id:id, price:price, type:type }});
    logging.addEntry(new Date(),{"bid":{id:id, price:price, type:type }});
  }
  this.bid_won = function (id, price) {
    console.log("bid_won", id, price);
    log_data.spend_sum += price;
    log_data.wins += 1;
    log_data.entries.push({"win":{id:id,price:price}});
    logging.addEntry(new Date(),{"win":{id:id,price:price}});
  }
  this.nobid_requested = function () {
    console.log("nobid_requested");
    log_data.nobid_count += 1;
    log_data.entries.push({"nobid":{}});
    logging.addEntry(new Date(),{"nobid":{}});
  }

  this.data = function () {
    return log_data;
  }
  this.reset = function () {
   log_data ={ entries:[], wins:0, spend_sum:0, bid_sum:0, bid_count:0, nobid_count:0 };
  }

  this.setup = function(mongo_url, cb) {
    console.log("setup Logger");
    logging.setupLogging(mongo_url, cb);
  }
}



module.exports = new Logger();
